lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -121,12 +122,23 @@ def file_path
     end
 
     def sync(mediator)
-      @instance_data = mediator.get_instance(kubectl_resource_type, name)
+      @instance_data = mediator.get_instance(kubectl_resource_type, name, raise_on_404: true)
+    rescue KubernetesDeploy::Kubectl::ResourceNotFoundError
+      @disappeared = true if deploy_started?
+      @instance_data = {}
     end
 
     def after_sync
     end
 
+    def deleted?
+      @instance_data.dig('metadata', 'deletionTimestamp').present?
```

Comment:
> So this basically means that k8s has asked the pod to "please go away", but in theory the pod might still exist and the process might still be running, right? 

Yes. I should probably call this `terminating?` -- that's the terminology kubectl uses.

> What's the reasoning for not checking whether the pod has actually already been deleted? Because this will catch deletion requests earlier?

Yes. We've also seen a k8s bug in the past where resources would get stuck in the terminating state indefinitely, even after the underlying container was gone (though it hasn't been reported in recent version afaik). (Note that we are also checking that they have actually been deleted--if so `disappeared?` will be true)